core/station/impl/src/models/request_policy.rs

```diff
@@ -53,10 +53,29 @@ impl From<MatchError> for EvaluateError {
     }
 }
 
+// metadata related rules can only be applied to accounts
+pub fn validate_rule_for_specifier(
+    rule: &RequestPolicyRule,
+    specifier: &RequestSpecifier,
+) -> ModelValidatorResult<RequestPolicyError> {
+    match (rule, specifier) {
```

Comment:
How about separating the two conditions:
```
let is_metadata_rule = match rule {
  RequestPolicyRule::AllowListed | RequestPolicyRule::AllowListedByMetadata(_) => true,
  _ => false,
};
let is_transfer = match specifier {
  RequestSpecifier::Transfer(_) => true,
  _ => false,
};
if is_metadata_rule && !is_transfer {
  Err(RequestPolicyError::InvalidRuleForSpecifier {
    rule: rule.to_string(),
    specifier: specifier.to_string(),
  })
} else {
  Ok(())
}
```

Reply:
I see. However, the risk with matching pairs is that we can easily miss some combinations in the product.

Reply:
I removed the tuple match and instead we're just matching for the rule exhaustively. Then the Transfer-only condition is simpler. What do you think? We could still list all request operations to not use the wildcard, maybe it's worth it?

Reply:
The current version looks good; I'd still remove the wildcard though.

Reply:
I have removed the wildcard.